mongo/single_result.go

```diff
@@ -33,12 +33,12 @@ type SingleResult struct {
 // will be returned. If there were no returned documents, ErrNoDocuments is
 // returned.
 func (sr *SingleResult) Decode(v interface{}) error {
-	if sr.reg == nil {
+	if sr.err != nil {
+		return sr.err
+	} else if sr.reg == nil {
```

Comment:
Let's just make this a regular `if`, no need to have it be an `else if`.
```suggestion
	}
	if sr.reg == nil {
```

Reply:
done